packages/askui-nodejs/src/utils/analytics/installation-timestamp.ts

```diff
@@ -0,0 +1,48 @@
+import fs from 'fs';
+import path from 'path';
+import { logger } from '../../lib';
+import { InstallationTimestampCreateError } from './installation-timestamp-create-error';
+import { InstallationTimestampGetError } from './installation-timestamp-get-error';
+
+export abstract class InstallationTimestamp {
+  private static fileName = path.resolve(__dirname, 'install-timestamp');
+
+  private static value?: Date | null;
+
+  static async create(): Promise<void> {
+    const timestamp = new Date().toISOString();
+    return new Promise((resolve, reject) => {
+      fs.writeFile(InstallationTimestamp.fileName, timestamp, { encoding: 'utf-8' }, (err) => {
+        if (err) {
+          reject(new InstallationTimestampCreateError(err));
+        } else {
+          resolve();
+        }
+      });
+    });
+  }
+
+  static async get(): Promise<Date | null> {
+    if (InstallationTimestamp.value === undefined) {
+      try {
+        InstallationTimestamp.value = await this.getFromFile();
+      } catch (err) {
+        InstallationTimestamp.value = null;
+        logger.error(err);
+      }
+    }
+    return InstallationTimestamp.value;
+  }
+
+  private static async getFromFile(): Promise<Date> {
+    return new Promise((resolve, reject) => {
+      fs.readFile(InstallationTimestamp.fileName, 'utf-8', (err, isoDateStr) => {
+        if (err) {
+          reject(new InstallationTimestampGetError(err));
+        } else {
+          resolve(new Date(isoDateStr));
```

Comment:
should we also check if isoDateStr is valid?

Reply:
It doesn't make sense from my perspective. It is unlikely that a user is going to mess with the contents of that file except people trying to misuse our api but we cannot prevent them from sending us bad data at this stage anyway. Let's say someone really changed it by accident, then, If the string was really not interpretable as a date, an error would be raised that would be caught and we would not get an installation timestamp.